### ISSUE TITLE
Recording of iterative message sends is wrong/counterintuitive

### ISSUE THREAD

**Issue by LinqLover** [hpi-swa/MessageSendRecorder#5]
Steps to reproduce:
```smalltalk
m := Morph new.
3 timesRepeat: [m addMorph: Morph new].

MessageSendRecorder new
	watchPackageNamed: #Morphic;
	setMessageSend: [m submorphsDo: [:n | n show]];
	record;
	browse.
```

Output:
![image](https://user-images.githubusercontent.com/38782922/92711936-44322f80-f359-11ea-8a26-68947cbe4de0.png)

The representation is counterintuitive. When using `#recordAllObjects`, the argument `n` is not recorded either.

Rather, I would expect to see three children of the `Morph>>submorphsDo:` node, one for each invocation of the `[] in UndefinedObject>>DoIt` block.

**Comment by LinqLover** [comment]
Unfortunately, the number of nested nodes does not even correspond with the number of iterations but the number of calls from the iterator block ...

![image](https://user-images.githubusercontent.com/38782922/92713980-e6531700-f35b-11ea-9a4f-da53299cd1c7.png)

**Comment by marceltaeumel** [comment]
Interesting. Well, it is not possible to catch neither block evaluation nor its return. That []-record is created manually. You discovered the drawback of that.

See:
https://github.com/hpi-swa/MessageSendRecorder/blob/master/packages/MessageSendRecorder.package/MessageSendRecorder.class/instance/recordSendId.method.receiver.arguments.context..st

Would be nice to also simulate the block return somehow.

**Comment by LinqLover** [comment]
Hm, in theory, this should be solved by checking `#outerCode` of the right `CompiledBlock`, shouldn't it? And does MSR support the V3.X bytecode set all?

**Comment by marceltaeumel** [comment]
I don't think that I use SistaV1-specific calls. should work with V3PlusClosures, too.

**Comment by LinqLover** [comment]
But this refers to SistaV1 explicitly:

https://github.com/hpi-swa/MessageSendRecorder/blob/befc38911ab66cc810aaa03539ae7b6d8374c0b3/packages/MessageSendRecorder.package/MessageSendRecorder.class/instance/recordSendId.method.receiver.arguments.context..st#L10

In V3PlusClosures, we don't have `CompiledBlock`s, so `aContext method` will always equal `closure homeMethod`, correctly?

**Comment by marceltaeumel** [comment]
Correct. So, where is the issue? :-) Just the comment refers to SistaV1, not the code.

**Comment by LinqLover** [comment]
FWIW, here is another erroneous record that assumes recursion instead of method calls:

![image](https://user-images.githubusercontent.com/38782922/109696927-e0612480-7b8d-11eb-9691-f3373b0caaa0.png)

Recorded watching the packages `#(Metacello Gofer)` with `#recordAllObjects`.
